lib/src/kargs.rs

```diff
@@ -229,7 +229,8 @@ match-architectures = ["x86_64", "aarch64"]
         .to_string();
         let parsed_kargs = parse_kargs_toml(&file_content, sys_arch).unwrap();
         assert_eq!(parsed_kargs, ["console=tty0", "nosmt"]);
-        std::env::set_var("ARCH", "aarch64");
+        // TODO: Audit that the environment access only happens in single-threaded code.
```

Comment:
It definitely doesn't, rust runs unit tests in multiple threads by default. We need to stop setting this environment variable in the tests at all. At a quick look it's not clear we need this line of code, shouldn't we be just doing `let sys_arch = "aarch64"`?

Reply:
Yeah it doesn't even reference the environment variable anywhere.  No idea what that was ever about.  I'll go open a separate PR against main to fix just that.

Reply:
> Yeah it doesn't even reference the environment variable anywhere. No idea what that was ever about. I'll go open a separate PR against main to fix just that.

#1132